core/common/buffer_or_view.hpp

```diff
@@ -88,11 +102,11 @@ namespace kagome::common {
 
     template <typename T, typename = AsSpan<T>>
     friend bool operator==(const BufferOrView &l, const T &r) {
-      return l.view() == Span{r};
+      return std::equal(l.view().begin(), l.view().end(), Span{r}.begin());
```

Comment:
Bug, out of bound, use `equal(b1, e1, b2, e2)`.

Why not use `libp2p::BytesIn{left} == libp2p::BytesIn{right}`?